recursive_merge_sort.py

```diff
@@ -0,0 +1,53 @@
+import timeit
+
+
+def merge_sort(a_list):
+    if len(a_list) <= 1:
+        return a_list
+    else:
+        return merge(merge_sort(a_list[:len(a_list) / 2]),
+                     merge_sort(a_list[len(a_list) / 2:]))
+
+
+def merge(left, right):
+    new_list = []
+    left_idx, right_idx = 0, 0
+    while left_idx < len(left) and right_idx < len(right):
+        if left[left_idx] < right[right_idx]:
+            new_list.append(left[left_idx])
+            left_idx += 1
+        else:
+            new_list.append(right[right_idx])
+            right_idx += 1
+
+    while left_idx < len(left):
+        new_list.append(left[left_idx])
+        left_idx += 1
```

Comment:
Is there a reason to continue iterating here?  Could you not just extend the new_list with the remainder of left or right?